cmd/syft/internal/ui/capture.go

```diff
@@ -0,0 +1,65 @@
+package ui
+
+import (
+	"io"
+	"os"
+	"time"
+
+	"github.com/anchore/syft/internal/log"
+)
+
+// Capture is a utility to capture output written to an os.File, particularly suited for capturing output of
+// os.Stdout or os.Stderr. Usage is fairly straightforward: the return value is a function which stops the current
+// capture and restores the previous file. For example:
+//
+//	restore := ui.Capture(&os.Stdout, writer)
+//	// here, stdout will be captured and redirected to the provided writer
+//	restore() // block until the output has all been sent to the writer and restore the original os.Stdout
+func Capture(target **os.File, writer io.Writer) (close func()) {
```

Comment:
nit: this is internal so grain of salt... changing this to 
```suggestion
func CaptureStdout(writer io.Writer) (close func()) {
```
would make this harder to misuse. You could still keep the internal `capture` and pivot tests against that function.